Fill.py

```diff
@@ -45,19 +49,41 @@ def sweep_from_pool():
             for i, location in enumerate(locations):
                 if (not single_player_placement or location.player == item_to_place.player) \
                         and location.can_fill(maximum_exploration_state, item_to_place, perform_access_check):
-                    spot_to_fill = locations.pop(i) # poping by index is faster than removing by content,
+                    # poping by index is faster than removing by content,
+                    spot_to_fill = locations.pop(i)
                     # skipping a scan for the element
                     break
 
             else:
-                # we filled all reachable spots. Maybe the game can be beaten anyway?
-                unplaced_items.append(item_to_place)
-                if world.accessibility[item_to_place.player] != 'minimal' and world.can_beat_game():
-                    logging.warning(
-                        f'Not all items placed. Game beatable anyway. (Could not place {item_to_place})')
-                    continue
-                raise FillError(f'No more spots to place {item_to_place}, locations {locations} are invalid. '
-                                f'Already placed {len(placements)}: {", ".join(str(place) for place in placements)}')
+                # we filled all reachable spots.
+                # try swaping this item with previously placed items
+                for(i, location) in enumerate(placements):
+                    placed_item = location.item
+                    location.item = None
+                    placed_item.location = None
+                    swap_state = sweep_from_pool(base_state, itempool)
+                    if (not single_player_placement or location.player == item_to_place.player) \
+                            and location.can_fill(swap_state, item_to_place, perform_access_check):
+                        # Add this item to the exisiting placement, and 
+                        # add the old item to the back of the queue
+                        spot_to_fill = placements.pop(i)
+                        reachable_items.setdefault(placed_item.player, []).append(placed_item)
+                        itempool.append(placed_item)
```

Comment:
This can cause an infinite loop if two items can both be exclusively placed in one location and not another. I'm working on a way to break that loop.